features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/data/APIClient.js

```diff
@@ -134,14 +134,44 @@ class APIClient {
 
     _getRequestInterceptor() {
         return (request) => {
-            AuthManager.refreshTokenOnExpire(request, this.environment);
+            // todo: sync token checking logic
+            const existingUser = AuthManager.getUser(this.environment.label);
+            if (!existingUser) {
+                console.log('User not found. Token refreshing failed.');
+                return request;
+            }
+            const existingToken = AuthManager.getUser(this.environment.label).getPartialToken();
+            const refToken = AuthManager.getUser(this.environment.label).getRefreshPartialToken();
+            if (existingToken && (existingToken !== 'undefined')) {
+                request.headers.authorization = 'Bearer ' + existingToken;
+                return request;
+            } else {
+                console.log('Access token is expired. Trying to refresh.');
+                if (!refToken) {
+                    console.log('Refresh token not found. Token refreshing failed.');
+                    return request;
+                }
+            }
+            const promise = new Promise((resolve, reject) => {
+                AuthManager.refresh(this.environment)
+                    .then(res => res.json())
+                    .then(() => {
+                        request.headers.authorization = 'Bearer '
+                            + AuthManager.getUser(this.environment.label).getPartialToken();
```

Comment:
I think this will still return the old partial token , You might need to call `setPartialToken` in the current user before getting the partial token

Reply:
Since it is a cookie, values get saved automatically once the response is received. Do we have to save it again?

Reply:
Ahh we are setting the cookie using set-cookie headers here ?
But there is a catch here, If the UI is running in a different host to where jaggery services are running, this approach will break.